DiffusionPreprocessing/scripts/DiffusionToStructural.sh

```diff
@@ -79,8 +79,9 @@ ${FSLDIR}/bin/imcp "$WorkingDirectory"/"$regimg"2T1w_restore "$RegOutput"
 ${FSLDIR}/bin/fslmaths "$T1wRestoreImage".nii.gz -mul "$WorkingDirectory"/"$regimg"2T1w_restore.nii.gz -sqrt "$QAImage"_"$regimg".nii.gz
 
 #Generate 1.25mm structural space for resampling the diffusion data into
-${FSLDIR}/bin/flirt -interp spline -in "$T1wRestoreImage" -ref "$T1wRestoreImage" -applyisoxfm ${DiffRes} -out "$T1wRestoreImage"_${DiffRes}
-${FSLDIR}/bin/applywarp --rel --interp=spline -i "$T1wRestoreImage" -r "$T1wRestoreImage"_${DiffRes} -o "$T1wRestoreImage"_${DiffRes}
+${FSLDIR}/bin/flirt -interp spline -in "$T1wRestoreImage" -ref "$T1wRestoreImage" -applyisoxfm ${DiffRes} -out "$WorkingDirectory"/T1w_acpc_dc_restore_${DiffRes}
```

Comment:
What is the need for the changes here?  Also, even if the output is going into the "$WorkingDirectory" we shouldn't be hard-coding the file name as "T1w_acpc_dc_restore".  It should respect/use the variables defined internally in the script.

Reply:
We don't want this change.  The T1w volume at DWI resolution is an expected packaged output.  It needs to be resampled with splines which did not originally work when using flirt and -applyisoxfm (hence the need for two lines), but perhaps that bug has since been solved.  

I wasn't expecting changes to the registration code (which was most of what I wrote for this pipeline).  Are there more changes here?

Reply:
@glasserm  I think the way that I inserted my comment gave you an incomplete view of the overall change proposed by @MichielCottaar.  You should look at the full set of proposed changes, which I still don't understand the purpose of.  There is still an 'applywarp', but the output is going into the "WorkingDirectory", only to get immediately moved to the location/name of the file created in the original version.

Reply:
Okay, so then I also don't understand the purpose of that.

Reply:
This change was made, so that you can safely run the diffusion preprocessing pipeline in parallel for the different shims before merging the output. For all the rest of the preprocessing code it is irrelevant whether the preprocessing pipeline for the different shims are run in sequence or in parallel, because the have different $WorkingDirectory. However, in this part of the code the script writes to the general T1w directory and hence the separate runs of the DiffusionToStructural.sh will actually write to the same file. 

In the original version $T1wRestoreImage temporarily contains the result from the flirt command and then the results from the applywarp command. If the preprocessing pipelines for the different shims are run in parallel, this means that at no point in the script can we be certain which of these files is actually stored on disk (as the other parallel script might at any point run the first flirt command). The new version ensures that the file stores on disk is always the result from the applywarp command, whether the scripts are run in parallel or sequence.

In practice this is very unlikely to lead to a bug, but if it does it will be very hard to find the origin of the bug (as it will depend on the timings of parallel scripts and hence not be reproducible). So it seemed like a good idea to make these changes to ensure that this problem never arises...

Reply:
Ok, my concern was indeed because I did not see that the file was still being made.